internal/rest/avi_obj_vsvip.go

```diff
@@ -95,8 +95,19 @@ func (rest *RestOperations) AviVsVipBuild(vsvip_meta *nodes.AviVSVIPNode, cache_
 			}
 		}
 
-		// Override the Vip in VsVip tto bring in updates, keeping everything else as is.
+		if !vsCache.EnableRhi && vsvip_meta.BGPPeerLabels != nil && len(*vsvip_meta.BGPPeerLabels) > 0 {
+			err = fmt.Errorf("To use selective vip advertisement, %s VS must advertise vips via BGP. Please recreate the VS.", vsCache.Name)
+			utils.AviLog.Error(err)
+			return nil, err
+		}
+
+		if vsvip_meta.BGPPeerLabels != nil {
```

Comment:
if enableRHI is `true` and then someone updates the BGPPeerLabels, this code assumes that the BGPPeerLabels can be updated on the vsvip using a PUT call.  I guess this is the only exception in vsvip PUT call with RHI enabled?

Reply:
yes, labels can be updated freely.